cmd/install/assets/hypershift-operator/hypershift.openshift.io_hostedclusters.yaml

```diff
@@ -7550,38 +7547,44 @@ spec:
                         default: default
                         description: |-
                           ResourceGroupName is the name of an existing resource group where all cloud resources created by the Hosted
-                          Cluster are to be placed.
+                          Cluster are to be placed. The resource group is expected to exist under the same subscription as SubscriptionID.
+
+
                           In ARO HCP, this will be the managed resource group where customer cloud resources will be created.
 
 
+                          Resource group naming requirements can be found here: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ResourceGroup.Name/.
+
+
                           Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
                                    ResourceGroupName is <resourceGroupName>.
+                        pattern: ^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$
                         type: string
                         x-kubernetes-validations:
                         - message: ResourceGroupName is immutable
                           rule: self == oldSelf
-                      securityGroupName:
+                      securityGroupID:
                         description: |-
-                          SecurityGroupName is the name of an existing security group on SubnetName. This field is provided as part of the
-                          configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM).
-
-
-                          Example: if your Network Security Group ID is /subscriptions/<subscriptionID>/resourcegroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<securityGroupName>,
-                                   your SecurityGroupName is <securityGroupName>
+                          SecurityGroupID is the ID of an existing security group on the SubnetID. This field is provided as part of the
+                          configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM). This security group is
+                          expected to exist under the same subscription as SubscriptionID and in the same resource group VnetID is located
+                          in.
                         type: string
                         x-kubernetes-validations:
-                        - message: SecurityGroupName is immutable
+                        - message: SecurityGroupID is immutable
```

Comment:
Something I think we can live with later, but I'm sure we're going to get an ask from a customer at some point to allow updating it because someone accidentally deleted their security group.

Reply:
I'm happy to make this mutable now, is this supported in ARO Classic today?

Reply:
I think to make it mutable requires updates within the hosted cluster to ensure the cloud provider config used by CCM is correctly updated.  

the openshift installer doesn't handle this today and doesn't plumb your own NSG into the cloud provider config. 

How does the cloud provider config get laid down on the hosted cluster?  Or is it all management side?  I'd be tempted to ask Jerome and make this a separate item to not increase scope of this PR.

Reply:
It's plumbed automatically, thanks for the link in Slack :)

https://github.com/openshift/hypershift/blob/021003d7138a79f59165875ed5a71dfe6bc2e7b7/control-plane-operator/controllers/hostedcontrolplane/cloud/azure/providerconfig.go#L61

Let's ask Jerome and open a separate card as a result.  i'm assuming we'll want to be able to mutate this in case customer accidentally deletes their NSG.  Probably low priority.

Reply:
Following up with Jerome here - https://redhat-internal.slack.com/archives/C05Q232L29E/p1715114219874579?thread_ts=1715114133.236519&cid=C05Q232L29E.